src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/OrderedDictionary.cs

```diff
@@ -0,0 +1,981 @@
+        /// </summary>
+        /// <param name="collection">The <see cref="IEnumerable{KeyValuePair{TKey, TValue}}" /> whose elements are copied to the new <see cref="OrderedDictionary{TKey, TValue}" />.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="collection" /> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="collection" /> contains one or more duplicate keys.</exception>
+        public OrderedDictionary(IEnumerable<KeyValuePair<TKey, TValue>> collection)
+            : this(collection, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrderedDictionary{TKey, TValue}" /> class that contains elements copied from the specified <see cref="IEnumerable{KeyValuePair{TKey, TValue}}" /> and uses the specified <see cref="IEqualityComparer{T}" />.
+        /// </summary>
+        /// <param name="collection">The <see cref="IEnumerable{KeyValuePair{TKey, TValue}}" /> whose elements are copied to the new <see cref="OrderedDictionary{TKey, TValue}" />.</param>
+        /// <param name="comparer">The <see cref="IEqualityComparer{T}" /> implementation to use when comparing keys, or null to use the default <see cref="EqualityComparer{T}" /> for the type of the key.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="collection" /> is null.</exception>
+        /// <exception cref="ArgumentException"><paramref name="collection" /> contains one or more duplicate keys.</exception>
+        public OrderedDictionary(IEnumerable<KeyValuePair<TKey, TValue>> collection, IEqualityComparer<TKey> comparer)
+            : this((collection as ICollection<KeyValuePair<TKey, TValue>>)?.Count ?? 0, comparer)
+        {
+            if (collection == null)
+            {
+                throw new ArgumentNullException(nameof(collection));
+            }
+
+            foreach (KeyValuePair<TKey, TValue> pair in collection)
+            {
+                Add(pair.Key, pair.Value);
+            }
+        }
+
+        /// <summary>
+        /// Adds the specified key and value to the dictionary as an O(1) operation.
+        /// </summary>
+        /// <param name="key">The key of the element to add.</param>
+        /// <param name="value">The value of the element to add. The value can be null for reference types.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="key" /> is null.</exception>
+        /// <exception cref="ArgumentException">An element with the same key already exists in the <see cref="OrderedDictionary{TKey, TValue}" />.</exception>
+        public void Add(TKey key, TValue value) => TryInsert(null, key, value, InsertionBehavior.ThrowOnExisting);
+
+        /// <summary>
+        /// Removes all keys and values from the <see cref="OrderedDictionary{TKey, TValue}" />.
+        /// </summary>
+        public void Clear()
+        {
+            if (_count > 0)
```

Comment:
BTW, in DictionarySlim I made Clear() reset the capacity also. Dictionary retains the arrays as you have done. I guess it depends on the use case, whether you are likely to need that capacity soon enough again.

BTW did you mean to increment version?

Reply:
I was going to fix it - but your branch is still read only 😱 .  for future reference you can check this box if you want to accept changes to a PR:
![image](https://user-images.githubusercontent.com/6385855/48750953-c3132100-ec36-11e8-88b4-76008dcdbc19.png)

Reply:
That's weird because the checkbox is checked for me.